python/lsst/ts/audiotrigger/read_serial_temp_scanner.py

```diff
@@ -0,0 +1,313 @@
+        elif new_data[self.fan_sensor] < self.fan_turn_off_temp:
+            self.log.info(f"Turning OFF fan, temperature: {new_data[self.fan_sensor]}")
+            await self.set_fan_off()
+
+        # Now for telemetry, do a rolling average of all 8 sensors
+        new_temperature = 0
+        for reading in new_data:
+            data = float(new_data[reading])
+            self.log.info(f"reading: {data}")
+            new_temperature += data
+        new_temperature = new_temperature / len(new_data)
+
+        # Handle initial data loading
+        if self.first_run:
+            self.first_run = False
+            for i in range(len(self.rolling_temperature)):
+                self.rolling_temperature[i] = new_temperature
+        else:
+            # Update rolling temperature windows
+            # freshest data starts at 0 index and goes towards max length index
+            for i in range(len(self.rolling_temperature) - 1, 0, -1):
+                self.rolling_temperature[i] = self.rolling_temperature[i - 1]
+
+        # Update freshest at index 0
+        self.log.info(f"New rolling temperature data logged: {new_temperature}")
+        await self.write_if_connected(f"TS: New: {new_temperature}")
+        self.rolling_temperature[0] = new_temperature
+        self.log.info("test after rolling temp")
+
+    async def get_data(self):
+        """Get incoming data."""
+        try:
+            in_waiting = await self.loop.run_in_executor(None, self.serial.inWaiting)
+            readings = await self.loop.run_in_executor(
+                None, functools.partial(self.serial.read, in_waiting)
+            )
+            readings = readings.decode("ISO-8859-1").rstrip()
+            # reads all data since last read
+            latest_reading = readings.split("\n")[-2]
+            # Reason for taking second to last reading rather
+            # than most recent is because most recent reading
+            # often doesn't have values from all sensors.
+            for reading in latest_reading.split(","):
+                try:  # Not all inputs will be used
+                    sensor_location, sensor_reading = reading.split("=")
+                    self.latest_data[self.sensor_dict[sensor_location]] = sensor_reading
+                    self.log.info(
+                        f"New data logged, reading: {sensor_reading}, location: {sensor_location}"
+                    )
+                except Exception:
+                    pass
+        except Exception as e:
+            await self.write_if_connected(
+                f"TS: Exception: Serial Temperature Scanner tried to get data, got exception instead: {e}"
+            )
+            self.log.warning(
+                f"Serial Temperature Scanner tried to get data, got exception instead: {e}"
+            )
+
+    async def serial_temperature_task(self):
+        """Get incoming data and publish through the server."""
+        # Read sensors, write to file, close file,
```

Comment:
I think this line may be removed since no writing to files takes place.

Reply:
You marked the comment as resolved but the reference to writing to file remains. Please update the comment such that it reflects what really is done.

Reply:
I removed the comment to writing a file.